File: Nian_Karl/QL_Nian/src/ql/ast/expression/Unary.java
package ql.ast.expression;

import ql.ast.value.Value;

public abstract class Unary extends Expression{
	
	private final Expression operand;
	
	public Unary(Expression operand){
		this.operand = operand;
	}
	
	public Expression getExpression(){
		return this.operand;
	}
	
	@Override
	public abstract Value<?> evaluate();
}

File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Expression/Unary/Unary.cs
using AST.Nodes.Interfaces;

namespace AST.Nodes.Expression.Unary
{
    public class Unary : ASTNode, IExpression
    {
        private readonly IExpression child;
        
        protected Unary(IExpression child, Representation.PositionInText pos)
            :base(pos)
        { this.child = child; }

        public IExpression GetChildExpression()
        { return child; }

        public virtual T Accept<T>(ASTVisitors.IVisitor<T> visitor)
        {
            return this.Accept(visitor);
        }
    }
}
